### PR TITLE
Documentation party

### PR DESCRIPTION
**Pull request by allthesignals**
This PR adds more documentation to the app. It still needs a lot more documentation, specifically in w/r/t the rows.

### CODE REVIEW

File: app/mixins/downloadable.js [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
+ *    with the nested structure on the controller, necessary for profile data display, downloading
+ *    and charts.
+ */
 export default Mixin.create({
-  selection: service(),
-
+  /**
+   * EmberJS Route beforeModel hook.
+   * Here this is used to unload the store to avoid the issue with duplicate ids.
+   * TODO: There are duplicates because id is based arbitrarily on the array index.
```
Comment (julialucyhogan) [inline app/mixins/downloadable.js]: does this mean that if we had different IDs, we could cache all the profiles for a given session and not have to redownload when selecting back & forth btwn multiple profiles? we should add this to a todo ticket :)

File: app/models/row.js [hunk 2]
```diff
@@ -2,81 +2,318 @@ import { alias } from '@ember/object/computed';
...
+
+  /**
+   * Category name, used to group together rows
+   * e.g. sexAndAge (as part of profile "demographic"), placeOfBirth (as part of profile "social")
+   */
+  category: DS.attr('string'),
+
+  /**
+   * Raw variable name, used to group together rows.
+   * e.g. sexAndAge (as part of profile "demographic"), placeOfBirth (as part of profile "social")
```
Comment (julialucyhogan) [inline app/models/row.js]: typo? this line seems copy pasta from above

File: app/models/row.js [hunk 3]
```diff
@@ -2,81 +2,318 @@ import { alias } from '@ember/object/computed';
...
+   */
   variable: DS.attr('string'), // e.g. Pop10t14 (as part of category "sexAndAge")
+
+  /**
+   * Lower-cased version of "variable"
```
Comment (julialucyhogan) [inline app/models/row.js]: opposite -- `variablename` is the regular cased version, `variable` is the lowercase version

File: app/routes/data.js [hunk 4]
```diff
@@ -1,7 +1,17 @@
 import Route from '@ember/routing/route';
 import layerGroups from '../layer-groups';
 
+/**
+ * The data route is responsible for statically importing layer groups
```
Comment (julialucyhogan) [inline app/routes/data.js]: what does 'displaying a layer groups information' mean? Are these for displaying the selected geo when viewing a profile? or for displaying the map used to create selections? or both?

File: app/routes/index.js [hunk 5]
```diff
@@ -3,9 +3,24 @@ import { inject as service } from '@ember/service';
 import { next } from '@ember/runloop';
 import { hash } from 'rsvp';
 
+/**
+ * The Index Route is responsible for pulling down layer groups for the map,
```
Comment (julialucyhogan) [inline app/routes/index.js]: which map?

File: app/mixins/downloadable.js (hunk 1 above)
Comment (allthesignals) [reply app/mixins/downloadable.js]: https://github.com/NYCPlanning/labs-factfinder/issues/711

@julialucyhogan

File: app/routes/data.js (hunk 4 above)
Comment (allthesignals) [reply app/routes/data.js]: @julialucyhogan I don't really know, the template has markup for looping over the layer-groups which are used in the _map_ when during the geo selection process: https://factfinder-staging.planninglabs.nyc/data#12.25/40.724/-73.9868.

Comment (allthesignals) [reply app/routes/data.js]: I think this should become a separate issue

**Pull request merged**